data-prepper-plugins/common/src/main/java/org/opensearch/dataprepper/plugins/s3keyindex/S3ObjectIndex.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.s3keyindex;
+
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Set;
+import java.util.TimeZone;
+import java.util.UUID;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Class responsible for creation of s3 key pattern based on date time stamp
+ */
+public class S3ObjectIndex {
+
+    private static final String TIME_PATTERN_STARTING_SYMBOLS = "\\${";
+
+    // For matching a string that begins with a "${" and ends with a "}".
+    // For a string like "data-prepper-${yyyy-MM-dd}", "${yyyy-MM-dd}" is matched.
+    private static final String TIME_PATTERN_REGULAR_EXPRESSION = "\\$\\{.*?\\}";
+
+    // For matching a string enclosed by "%{" and "}".

```

Comment:
The comment says "%{" but the code has "${". If we want to be consistent with Opensearch Sink, "%{" is more accurate. Not sure what was decided but make sure at the minimum comment and the code are consistent. I think it would be better to "%{" just like OpenSearch Sink for consistency across different sink configutations. But if there is a good reason to use "${", I am OK.

Reply:
Modified the comment as "%{".